osm2gtfs/creators/et_addisababa/trips_creator_et_addisababa.py

```diff
@@ -0,0 +1,133 @@
+
+        lines = data.routes
+        for route_ref, line in sorted(lines.iteritems()):
+            if not isinstance(line, Line):
+                continue
+            print("Generating schedule for line: " + route_ref)
+
+            flex_flag = None
+            if 'route_master' in line.tags and line.tags['route_master'] == "light_rail":
+                route_type = "Tram"
+                route_suffix = " (Light Rail)"
+            elif 'route_master' in line.tags and line.tags['route_master'] == "share_taxi":
+                route_type = "Bus"
+                route_suffix = " (Minibus)"
+                flex_flag = 0
+            else:
+                route_type = "Bus"
+                route_suffix = ""
+
+            line_gtfs = feed.AddRoute(
+                short_name=str(line.route_id),
+                long_name=line.name + route_suffix,
+                # we change the route_long_name with the 'from' and 'to' tags
+                # of the last route as the route_master name tag contains
+                # the line code (route_short_name)
+                route_type=route_type,
+                route_id=line.osm_id)
+            line_gtfs.agency_id = feed.GetDefaultAgency().agency_id
+            line_gtfs.route_desc = ""
+            line_gtfs.route_color = "1779c2"
+            line_gtfs.route_text_color = "ffffff"
+
+            route_index = 0
+            itineraries = line.get_itineraries()
+            for a_route in itineraries:
+                trip_gtfs = line_gtfs.AddTrip(feed)
+                trip_gtfs.shape_id = self._add_shape_to_feed(
+                    feed, a_route.osm_id, a_route)
+                trip_gtfs.direction_id = route_index % 2
+                route_index += 1
+
+                if a_route.fr and a_route.to:
+                    trip_gtfs.trip_headsign = a_route.to
+                    line_gtfs.route_long_name = a_route.fr.decode(
+                        'utf8') + " ↔ ".decode(
+                        'utf8') + a_route.to.decode('utf8') + route_suffix
+
+                DEFAULT_ROUTE_FREQUENCY = 30
+                DEFAULT_TRAVEL_TIME = 120
+
+                frequency = None
+
+                ROUTE_FREQUENCY = DEFAULT_ROUTE_FREQUENCY
+
+                if "interval" in a_route.tags:
+                    frequency = a_route.tags['interval']
+                    try:
+                        ROUTE_FREQUENCY = time_string_to_minutes(frequency)
+                        if not ROUTE_FREQUENCY > 0:
+                            print("frequency is invalid for route_master " + str(
+                                line.osm_id))
+                    except (ValueError, TypeError) as e:
+                        print("frequency not a number for route_master " + str(
+                                line.osm_id))
+
+                trip_gtfs.AddFrequency(
+                    "05:00:00", "22:00:00", ROUTE_FREQUENCY * 60)
+
+                if 'duration' in a_route.tags:
```

Comment:
the `ci_abidjan` creator also use `duration` and `interval` tags for the same purpose :thinking: I wonder if / how we can mutualize the implementations to avoid code duplication.

Reply:
Interesting, how could we do this?